cmd/cfm-service/main.go

```diff
@@ -68,17 +68,23 @@ func main() {
 	defaultRedfishController := redfishapi.NewDefaultAPIController(defaultRedfishService)
 	api.AddRedfishRouter(ctx, router, defaultRedfishController)
 
-	// Discover devices before loading datastore
-	bladeDevices, errBlade := services.DiscoverDevices(ctx, defaultApiService, "blade")
-	hostDevices, errHost := services.DiscoverDevices(ctx, defaultApiService, "cxl-host")
-	// Add the discovered devices into datastore
-	if errBlade == nil && errHost == nil {
-		services.AddDiscoveredDevices(ctx, defaultApiService, bladeDevices, hostDevices)
-	}
-
 	// Load datastore
 	datastore.DStore().Restore()
 	data := datastore.DStore().GetDataStore()
+
+	// If there are any devices in the data store, skip discovery
+	if len(data.ApplianceData) == 0 && len(data.HostData) == 0 {
```

Comment:
Do we need to handle the case where there are 1 or more appliances, but none of the appliances have any blades in them?  Would this situation mean that we still do NOT want to do the discovery?

Reply:
I would like to check until blade level, if there are no blades or hosts, do discovery. Please check my new code.